railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -510,6 +510,45 @@ def generate_models_for_animals
         end
       end
 
+      test "db:schema:load:name doesn't purge the test database. The test schema is maintained when running tests" do
```

Comment:
So with this change, what do you need to do to update the test database?

Reply:
In most scenarios you won't have to do anything manually to update the test schema, it will be taken care of by Rails whenever a test is ran.

One of the only reason I can think of where you may need to update it manually (by running the `bin/rails db:test:prepare command`) is if you run a console in the test environment `RAILS_ENV=test bin/rails c`, just after the schema is updated. This I think is very edge case and I don't think would cause issues.

It's worth noting that loading all schemas `db:schema:load` also doesn't purge the tests database and AFAIK hasn't caused any issues for users.

Reply:
> you won't have to do anything manually to update the test schema, it will be taken care of by Rails whenever a test is ran.

But isn't this what this test is showing? That the test schema is not updated between test runs?

(It's very likely I'm not understanding it!)

Reply:
> That the test schema is not updated between test runs?

The opposite 😅 .

The schema is modified here https://github.com/rails/rails/blob/ef1ccde0d8825a5d0dc784e3185b206d5bd8d56c/railties/test/application/rake/multi_dbs_test.rb#L539-L541

and we check that running the test once more keep the test schema up to date 

https://github.com/rails/rails/blob/ef1ccde0d8825a5d0dc784e3185b206d5bd8d56c/railties/test/application/rake/multi_dbs_test.rb#L547-L548

----

I suppose the test name is confusing?

Reply:
Well it clearly confused me, but now that you've explained it to me and I've read it again, I am pretty sure you are right and the test is fine. 

Sorry for the confusion!